scipy/sparse/tests/test_construct.py

```diff
@@ -248,9 +248,27 @@ def test_eye(self):
         for m in [3, 5]:
             for n in [3, 5]:
                 for k in range(-5,6):
-                    assert_equal(construct.eye(m, n, k=k).toarray(), np.eye(m, n, k=k))
-                    if m == n:
-                        assert_equal(construct.eye(m, k=k).toarray(), np.eye(m, n, k=k))
+                    # scipy.sparse.eye deviates from np.eye here. np.eye will
+                    # create arrays of all 0's when the diagonal offset is
+                    # greater than the size of the array. For sparse arrays
+                    # this makes less sense, especially as it results in dia
+                    # arrays with negative diagonals. Therefore sp.sparse.eye
+                    # validates that diagonal offsets fall within the shape of
+                    # the array. See gh-18555.
+                    if abs(k) > m or k > n:
+                        pytest.raises(
+                            ValueError, match="Offset.*out of bounds"
+                        )
```

Comment:
I think you need to fun the function to get the ValueError

Reply:
Of course - nice catch!

I should've been suspicious when it didn't raise on the `match` locally, since I never get regex's right on my first attempt :). Fixed in ead4a42 by using `pytest.raises` as a context manager. This also highlighted a bit of tightening that was necessary on the conditional for determining when a diagonal is OOB. Thanks @dschult !